language/src/parser.rs

```diff
@@ -1040,6 +1040,23 @@ fn pstate_declaration(i: PInput) -> PResult<PStateDeclaration> {
     )(i)
 }
 
+/// A State Declaration is a given required state on a given resource
+#[derive(Debug, PartialEq)]
+pub struct PBlockDeclaration<'src> {
+    pub metadata: Vec<PMetadata<'src>>,
+    pub childs: Vec<PStatement<'src>>,
+}
+fn pblock_declaration(i: PInput) -> PResult<PBlockDeclaration> {
+    wsequence!(
+        {
+            childs: delimited_parser("{", |j| many0(pstatement)(j), "}");
```

Comment:
`|j| many0(pstatement)(j)` is equivalent to : `many0(pstatement)`

Reply:
that does not compile with the following error

 ```the trait bound `impl std::ops::Fn<(nom_locate::LocatedSpan<&str, &str>,)>: std::marker::Copy` is not satisfied

the trait `std::marker::Copy` is not implemented for `impl std::ops::Fn<(nom_locate::LocatedSpan<&str, &str>,)>`rustc(E0277)
baseparsers.rs(196, 8): required by a bound in this
baseparsers.rs(202, 8): required by this bound in `parser::baseparsers::delimited_parser`
parser.rs(1052, 33): the trait `std::marker::Copy` is not implemented for `impl std::ops::Fn<(nom_locate::LocatedSpan<&str, &str>,)>```

I'll keep it like that atm

Reply:
OK :shrug: